deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java

```diff
@@ -133,42 +130,55 @@ public void runJob() {
 						}
 					}
 				}
+			default:
+				break;
 			}
 		}
 
-		// delete project staging folder, if exists/applicable
-		String sfuris = getDepositStatus().get(DepositField.stagingFolderURI.name());
-		if (sfuris != null && sfuris.trim().length() > 0) {
-			SharedStagingArea area = null;
-			URI sfuri = null;
+		// Cleanup files and directories specifically requested be cleaned up by an earlier job
+		NodeIterator it = m.listObjectsOfProperty(dprop(m, DepositRelationship.cleanupLocation));
+		while (it.hasNext()) {
+			RDFNode n = it.nextNode();
+			String stagingLoc = n.asLiteral().getString();
+			URI stagingUri = URI.create(stagingLoc);
+
+			if(!tagPattern.matches(stagingUri)) continue;
+
+			SharedStagingArea area = stages.findMatchingArea(stagingUri);
+			if (area == null) {
+				LOG.error("Cannot find staging area for URI: " + stagingUri.toString());
+				break;
+			}
+			if (!area.isConnected()) {
+				stages.connect(area.getURI());
+			}
+
+			URI storageUri = null;
 			try {
-				sfuri = new URI(sfuris);
-				area = stages.findMatchingArea(sfuri);
-			} catch (URISyntaxException ignore) {
+				storageUri = area.getStorageURI(stagingUri);
+			} catch (StagingException e) {
+				LOG.error("Could not resolve storage URI: {}", stagingUri.toString(), e);
 			}
-			if (area != null) {
-				if (CleanupPolicy.DELETE_INGESTED_DEPOSIT_FOLDERS.equals(area
-						.getIngestCleanupPolicy())) {
-					if (!area.isConnected()) {
-						stages.connect(area.getURI());
-					}
-					if(area.isConnected()) {
+			CleanupPolicy p = area.getIngestCleanupPolicy();
+			switch (p) {
+			case DELETE_INGESTED_FILES:
+			case DELETE_INGESTED_FILES_EMPTY_FOLDERS:
+				File cleanupFile = new File(storageUri);
+				if (cleanupFile.exists()) {
+					if (cleanupFile.isDirectory()) {
 						try {
-							URI depositStagingFolderURI = area.getStorageURI(sfuri);
-							FileUtils.deleteDirectory(new File(depositStagingFolderURI));
-							LOG.debug("deleted {}", new File(depositStagingFolderURI));
-						} catch (StagingException e) {
-							LOG.error("Cannot obtain storage location for deposit staging folder URI: "
-									+ sfuris, e);
+							FileUtils.deleteDirectory(cleanupFile);
 						} catch (IOException e) {
-							LOG.error("Cannot delete deposit staging folder: "
-									+ sfuris, e);
+							LOG.error("Cannot delete deposit directory: {}",
+									getDepositDirectory().getAbsolutePath(), e);
```

Comment:
Is this going to give us the correct error message? Didn't we try to delete the "cleanupFile"?